src/coreclr/src/vm/reflectioninvocation.cpp

```diff
@@ -2176,71 +2007,237 @@ lExit: ;
 }
 FCIMPLEND
 
-//*************************************************************************************************
-//*************************************************************************************************
-//*************************************************************************************************
-//      ReflectionSerialization
-//*************************************************************************************************
-//*************************************************************************************************
-//*************************************************************************************************
-FCIMPL1(Object*, ReflectionSerialization::GetUninitializedObject, ReflectClassBaseObject* objTypeUNSAFE) {
-    FCALL_CONTRACT;
+/*
+ * Given a RuntimeType, queries info on how to instantiate the object.
+ * pRuntimeType - [required] the RuntimeType object
+ * pRefType - [required] the RuntimeType
+ * ppfnAllocator - [required, null-init] fnptr to the allocator
+ *                 mgd sig: void* -> object
+ * pvAllocatorFirstArg - [required, null-init] first argument to the allocator
+ *                       (normally, but not always, the MethodTable*)
+ * fUnwrapNullable - if true and type handle is Nullable<T>, queries info for T
+ * ppfnCtor - [optional, null-init] the instance's parameterless ctor,
+ *            mgd sig object -> void, or null if no parameterless ctor exists
+ * pfCtorIsPublic - [optional, null-init] whether the parameterless ctor is public
+ * ppMethodTable - [required, null-init] the MethodTable* for the queried type
+ * ==========
+ * This method will not run the type's static ctor or instantiate the type.
+ */
+void QCALLTYPE RuntimeTypeHandle::GetActivationInfo(
+    QCall::ObjectHandleOnStack pRuntimeType,
+    PCODE* ppfnAllocator,
+    void** pvAllocatorFirstArg,
+    BOOL fUnwrapNullable,
```

Comment:
Would it be simpler to imply the `fUnwrapNullable` behavior based on `fGetUninitializedObject` instead of passing it explicitly? I do not expect we will ever need the flexibility that this additional parameter provides.

Reply:
That'll probably work. Related question: do you know a cleaner way to pass the `RuntimeType` instance? Using `QCall::ObjectHandleOnStack` seems like a bit of a kludge, resulting in needing to put the GC into cooperative mode in a few scopes inside this function.

Reply:
I cannot think about a better to do this with the current design of the COM activation.